frontend/public/components/modals/delete-modal.tsx

```diff
@@ -34,6 +34,7 @@ const DeleteModal = withHandlePromise((props: DeleteModalProps & HandlePromisePr
       : null;
 
     props.handlePromise(k8sKill(kind, resource, {}, {}, json), () => {
+      props.cleanupResource && props.callback();
```

Comment:
You don't need the extra boolean, just call the method if it exists.
```suggestion
      props.callback?.();
```